src/pricecypher/datasets.py

```diff
@@ -201,6 +203,7 @@ def get_transactions(
         request_data = {
             'aggregate': aggregate,
             'select_scopes': select_scopes,
+            'dataset_environment': dataset_environment,
```

Comment:
```suggestion
            'environment': dataset_environment,
```
The key in the request data must be exactly `environment` since that it what the Dataset Service will try to lookup in the request data. I'd suggest conditionally including this value though (see first comment).